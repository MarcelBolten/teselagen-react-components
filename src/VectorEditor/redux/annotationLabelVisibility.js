//./caretPosition.js
import { createReducer } from "redux-act";
import createAction from "./utils/createMetaAction";
const visibilityInitialValues = {
  features: true,
  parts: true,
  cutsites: true
};

// ------------------------------------
// Actions
// ------------------------------------
export const annotationLabelVisibilityToggle = createAction(
  "annotationLabelVisibilityToggle"
);

// ------------------------------------
// Reducer
// ------------------------------------
export default createReducer(
  {
    [annotationLabelVisibilityToggle]: (state, payload) => {
      return {
        ...state,
        [payload]: !state[payload]
      };
    }
  },
  visibilityInitialValues
);
