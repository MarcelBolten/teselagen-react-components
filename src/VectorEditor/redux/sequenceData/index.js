import createAction from "../utils/createMetaAction";
import features from "./features";
import primers from "./primers";
import sequence from "./sequence";
import translations from "./translations";
import { combineReducers } from "redux";
import cleanSequenceData from "../../utils/cleanSequenceData";
import { createReducer } from "redux-act";
// export * from './sharedActionCreators';
export * from "./primers";
// export * from './features';
// export * from './sequence';
// export * from './circular';
export * from "./translations";

// ------------------------------------
// Actions
// ------------------------------------

const _updateSequenceData = createAction("SEQUENCE_DATA_UPDATE");
export const updateSequenceData = function(seqData, ...rest) {
  return _updateSequenceData(cleanSequenceData(seqData), ...rest);
};

// ------------------------------------
// Reducer
// ------------------------------------
export default function(state, action) {
  var stateToPass = state;
  if (action.type === "SEQUENCE_DATA_UPDATE") {
    stateToPass = action.payload;
  }
  return combineReducers({
    primers,
    features,
    sequence,
    parts: (state = {}) => state,
    size: (state = {}) => state,
    circular: createReducer({}, true),
    translations,
    name: createReducer({}, ""),
    fromFileUpload: createReducer({}, false)
  })(stateToPass, action);
}
