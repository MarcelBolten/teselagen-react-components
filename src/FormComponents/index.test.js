// import each from 'lodash/each';
import {
  InputField,
  SelectField
  // FileUploadField,
  // AntFileUploadField,
  // DateInputField,
  // CheckboxField,
  // TextareaField,
  // EditableTextField,
  // NumericInputField,
  // RadioGroupField,
  // ReactSelectField,
} from "./index";
import React from "react";

// See README for discussion of chai, enzyme, and sinon
import { mount } from "enzyme";
// import sinon from 'sinon'

// In this file we're doing an integration test. Thus we need to hook up our
// form component to Redux and Redux-Form. To do that, we need to create the
// simplest redux store possible that will work with Redux-Form.
import { reducer as formReducer, reduxForm } from "redux-form";
import { createStore, combineReducers } from "redux";
import { Provider } from "react-redux";

describe("form components", function() {
  it("InputField", function() {
    const FormContainer = reduxForm({
      form: "testForm",
      validate: function(values) {
        const errors = {};
        if (!values.testField) {
          errors.testField = "Required";
        }
        return errors;
      }
    })(function(argument) {
      return (
        <div>
          <InputField name="testField" />
        </div>
      );
    });

    const mountedForm = mount(
      <Provider store={createStore(combineReducers({ form: formReducer }))}>
        <FormContainer />
      </Provider>
    );

    const input = mountedForm.find("input").first();
    expect(input).toHaveLength(1);
    // Our form component only shows error messages (help text) if the
    // field has been touched. To mimic touching the field, we simulate a
    // blur event, which means the input's onBlur method will run, which
    // will call the onBlur method supplied by Redux-Form.
    input.simulate("blur");

    expect(input).toHaveLength(1);
    input.simulate("change", { target: { value: "testValue" } });
    const { value } = input.props();
    expect(value).toEqual("testValue");
  });
  const nestedJson = { my: "nestedJson" };
  it("SelectField functions as expected", function() {
    const FormContainer = reduxForm({
      form: "testForm",
      validate: function(values) {
        const errors = {};
        if (!values.testField) {
          errors.testField = "Required";
        }
        return errors;
      }
    })(function(argument) {
      return (
        <div>
          <SelectField
            defaultValue={"firstVal"}
            options={[
              { value: "firstVal", label: "firstLabel" },
              { value: "testValue", name: "myOpt1" },
              { value: nestedJson, name: "anotherValuename" }
            ]}
            name="testField"
          />
        </div>
      );
    });
    const store = createStore(combineReducers({ form: formReducer }));
    const mountedForm = mount(
      <Provider store={store}>
        <FormContainer />
      </Provider>
    );

    const input = mountedForm.find("select").first();
    expect(input).toHaveLength(1);
    // Our form component only shows error messages (help text) if the
    // field has been touched. To mimic touching the field, we simulate a
    // blur event, which means the input's onBlur method will run, which
    // will call the onBlur method supplied by Redux-Form.
    input.simulate("blur");
    expect(store.getState().form.testForm.values.testField).toEqual("firstVal");
    expect(input.props().value).toEqual("firstVal");

    expect(input).toHaveLength(1);
    input.simulate("change", { target: { value: "anotherValue" } });
    expect(store.getState().form.testForm.values.testField).toEqual(
      "anotherValue"
    );

    input.simulate("change", { target: { value: JSON.stringify(nestedJson) } });
    expect(store.getState().form.testForm.values.testField).toEqual(nestedJson);
  });
});
