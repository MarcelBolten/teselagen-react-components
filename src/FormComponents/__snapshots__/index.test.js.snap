// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders a snapshot 1`] = `<renderBlueprintInput />`;

exports[`Component renders a snapshot 2`] = `<renderBlueprintCheckbox />`;

exports[`Component renders a snapshot 3`] = `<renderBlueprintTextarea />`;

exports[`Component renders a snapshot 4`] = `<renderBlueprintNumericInput />`;

exports[`Component renders a snapshot 5`] = `<renderBlueprintNumericRadio />`;

exports[`Component renders a snapshot 6`] = `<renderBlueprintNumericSelector />`;
